### PR TITLE
Blog post: Validating shapes

### PR DESCRIPTION
**Pull request by gaurav**
Here's the first of the "development plan" blog posts! I'm sure it could benefit from more editing and a better figure, but I wanted to see what it looked like in the blog layout. I'm not sure the image works: our current template expands it to fit the width of the page, so it ends up HUGE. But it looks better than having no image at all.

If you'd rather edit this in Google Docs, let me know and I'll move it back there -- otherwise feel free to leave comments here and I'll fix things as necessary!

### CODE REVIEW

**Comment by hlapp** [comment]
@gaurav the URLs already work with local testing, but you need to set `site.url` correctly in the config file. This is accomplished by enabling the `_config-dev.yml` config file to be used as well. Look at the script `serve-local.sh`. I recommend anyway that you use the Docker container as well.

**Comment by gaurav** [comment]
@hlapp Ah, got it! I haven't gotten the Docker container to work, but if I run `bundle exec jekyll serve --config _config-dev.yml`, then all the links work correctly. I'll pull out the URL relativization on my two pull requests.

**Comment by gaurav** [comment]
@hlapp Done!

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+These tools would help us with two main goals:
+
+ 1. To confirm the basic relationships we’d expect to see: for example, we might expect every Node in a phylogeny will be related phylogenetically to at least one other node in the phylogeny.
+ 
+ 2. For a particular phylogeny, we could identify aspects of the structure we would expect to find: for example, a particular phylogeny might show one node being ancestral to another. If the nodes in the OWL ontology don’t have the same relationship, then something is wrong.
+
+In terms of basic relationships, the most basic test is that every phylogeny should contain at least one Node (<tt>[CDAO:0000140](http://purl.obolibrary.org/obo/CDAO_0000140)</tt>). If it has more than one Node, then every Node should have either a [<tt>has_Child</tt>](http://purl.obolibrary.org/obo/CDAO_0000149) or [<tt>has_Sibling</tt>](https://github.com/hlapp/phyloref/blob/c2a1b813690e3afc78c2abdacab216e368b5c83e/phyloref.owl#L81-L85) with at least one other Node. Furthermore, no node should be a child of itself.
```
Comment (hlapp) [inline _posts/blog/2016-10-13-validating-shapes.md]: See above re: the use of `<tt/>`.

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+In terms of specifics, I’m going to look at encoding lineage lines (for example, *Evaniscus tibialis* is a child of <tt>Node_7</tt>, which is a child of <tt>Node_3</tt>, which is a child of <tt>Node_2</tt>, which is a child of <tt>Node_1</tt>). If I can come up with a compact representation of these, it may be a neat way to make sure that ontologies are being converted correctly.
```
Comment (hlapp) [inline _posts/blog/2016-10-13-validating-shapes.md]: Again, see above re: `<tt/>`.

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+layout: post
+title: Validating shapes
+author: mrvaidya
```
Comment (hlapp) [inline _posts/blog/2016-10-13-validating-shapes.md]: And rebase against master, too, obviously.

Comment (gaurav) [inline _posts/blog/2016-10-13-validating-shapes.md]: Done and done!

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+layout: post
+title: Validating shapes
+author: mrvaidya 
+modified:
+excerpt: How do you know your RDF data is sensible? You test it!
```
Comment (gaurav) [inline _posts/blog/2016-10-13-validating-shapes.md]: I like "sensible" as  reminder that we could have something that's well-formed and meets our expectations, but we later realise wasn't actually "sensible", i.e. didn't convey all the sense we wanted it to; the testing may mislead us but hopefully will eventually show us the way to a comprehensively sensible representation. I do like the idea of something stronger. Maybe "meaningful" or "completely described"?

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+tags: ["software testing", "development plan"]
```
Comment (gaurav) [inline _posts/blog/2016-10-13-validating-shapes.md]: I was hoping that the tags would be a categorizable system, and that some day we'll be able to go back and look at all our "development plans" in one place. I'm also planning to alternate between "development plans" post like this one, where we set out something we plan to do, and "development results" posts where we report back on what we found (so next week I'll write something up along the lines of "ShEx was kind of a pain but doable, but SHACL is great, and here's what our tests look like ...". I'd like "development plans" to be a specific sort of document -- one that lays out why we're going to do what we're going to do in the next couple of weeks. Does that make sense? If not, I'm happy to switch this to "development".

File: _posts/blog/2016-10-13-validating-shapes.md (hunk 4 above)
Comment (hlapp) [inline _posts/blog/2016-10-13-validating-shapes.md]: In my view _meaningful_ is stronger than _sensible_, so I'd use that. But if you have reasons to use _sensible_ then that's fine too.

File: _posts/blog/2016-10-13-validating-shapes.md (hunk 3 above)
Comment (hlapp) [inline _posts/blog/2016-10-13-validating-shapes.md]: But maybe you haven't pushed up yet? I'm still seeing _mrvaidya_ in the diff.

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
...gy. [One](https://github.com/gaurav/phylo2owl/blob/f9d614866f69611894af9edde0f71c4a72850ed0/tests/test_execute.py) makes sure that the tool executes without error. [Another set of tests](https://github.com/gaurav/phylo2owl/blob/f9d614866f69611894af9edde0f71c4a72850ed0/tests/test_owl_output.py) checks that an example phylogeny we’ve included in our repository (thanks for the liberal licensing, [Mullins, Kawada, Balhoff and Deans 2012](http://zookeys.pensoft.net/articles.php?id=3135)!) produces valid RDF/XML that is identical to a copy of expected result we’ve archived in our repository. However, this is both limited and naïve. We'll know that _something_ is up if the file it produces changed -- but a set of axioms can often be written to a file in ways that produces different sequences of bytes, yet represent the exact same or logically equivalent axioms. Figuring out by hand in such a situation what changed, and whether the change is neutral or not, can be time-consuming even for only a handful of input files, let alone many. We’re going to need something more powerful if we want to be able to check hundreds or even thousands of different phyloreferences every time we make changes.
```
Comment (gaurav) [inline _posts/blog/2016-10-13-validating-shapes.md]: I don't like the little grey box that our CSS adds around <code> blocks -- I thought just typesetting them differently using <tt> made them look pretty good. I'm happy to leave this blog post using <code>, and then maybe tweak how <code> is rendered in the future?

I have no idea why I had all the authors there! I've fixed that.

File: _posts/blog/2016-10-13-validating-shapes.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+To accomplish this, we’re investigating a set of software tools that can detect the presence or absence of expected (or unexpected) relationships within an RDF document. These include:
+
+ 1. [Shape Expressions (ShEx)](http://shex.io/) come with their own compact syntax for describing the “[shape](https://www.w3.org/2014/data-shapes/wiki/Main_Page)” of RDF data. It has Python, Scala and Javascript libraries.
```
Comment (gaurav) [inline _posts/blog/2016-10-13-validating-shapes.md]: Fixed.

**Comment by gaurav** [comment]
Thanks for the comments: I've pushed the changes now. Let me know what you think!

**Pull request merged**